### PR TITLE
Use a non-mutable metrics map.

### PR DESCRIPTION
**Pull request by jkinkead**
One step closer to immutable context class.

### CODE REVIEW

File: xrpc/src/main/java/com/nordstrom/xrpc/server/Router.java [hunk 1]
```diff
@@ -83,36 +84,35 @@ public Router(XConfig config, int maxPayload) {
...
-    meterNamesByStatusCode.put(HttpResponseStatus.NOT_FOUND, namePrefix + "notFound");
-    meterNamesByStatusCode.put(
-        HttpResponseStatus.TOO_MANY_REQUESTS, namePrefix + "tooManyRequests");
-    meterNamesByStatusCode.put(
-        HttpResponseStatus.INTERNAL_SERVER_ERROR, namePrefix + "serverError");
-
-    for (Map.Entry<HttpResponseStatus, String> entry : meterNamesByStatusCode.entrySet()) {
-      ctx.getMetersByStatusCode().put(entry.getKey(), metricRegistry.meter(entry.getValue()));
+  /** Adds a meter for all HTTP response codes to the given XrpcConnectionContext. */
+  private void configureResponseCodeMeters(XrpcConnectionContext.Builder contextBuilder) {
```
Comment (andyday) [inline xrpc/src/main/java/com/nordstrom/xrpc/server/Router.java]: Maybe rename this buildResponseCodeMeters

Comment (jkinkead) [reply xrpc/src/main/java/com/nordstrom/xrpc/server/Router.java]: `configure` isn't my favorite. `build` connotes a method that returns a new instance, to me; I went with the simple `add`.

File: xrpc/src/main/java/com/nordstrom/xrpc/server/Router.java [hunk 2]
```diff
@@ -83,36 +84,35 @@ public Router(XConfig config, int maxPayload) {
...
+    namesByCode.put(HttpResponseStatus.NO_CONTENT, "noContent");
+    namesByCode.put(HttpResponseStatus.BAD_REQUEST, "badRequest");
+    namesByCode.put(HttpResponseStatus.UNAUTHORIZED, "unauthorized");
+    namesByCode.put(HttpResponseStatus.FORBIDDEN, "forbidden");
+    namesByCode.put(HttpResponseStatus.NOT_FOUND, "notFound");
+    namesByCode.put(HttpResponseStatus.TOO_MANY_REQUESTS, "tooManyRequests");
+    namesByCode.put(HttpResponseStatus.INTERNAL_SERVER_ERROR, "serverError");
+
+    // Create the proper metrics containers.
+    String namePrefix = "responseCodes.";
```
[the remark this answers is not in the document]
Comment (jkinkead) [reply xrpc/src/main/java/com/nordstrom/xrpc/server/Router.java]: I realized this doesn't even need to be a local and inlined below.

**Comment by jkinkead** [comment]
Merge required; updating.

**Pull request merged**